docs/kernels.md

```diff
@@ -91,6 +116,40 @@ The image will be uploaded to Hub and can be use in a YAML file as
 `<your hub org>/kernel:4.9.33` or as `<your hub
 org>/kernel:4.9.33-<hash>`.
 
+### Modifying the Config
+Each series of kernels has a config file dedicated to it in [../kernel/](../kernel), e.g.
+[kernel.config-4.9.x](../kernel/kernel.config-4.9.x). To build a particular series of kernel:
+
+1. Create a separate `git` branch (not required but *strongly* recommended)
+2. Modify the appropriate `kernel.config`, e.g. `kernel.config-4.9.x`
+3. Run `make build_<series>` with appropriate arguments per this section, e.g. `make build_4.9.x ORG=foo HASH=bar`
+4. Create a `.yml`, build and test
+
+You can modify the config in one of two ways:
+
+* Manually, editing the config file
+* Using a standard config generator, like `menuconfig`
+
+Generally, you will manually edit a file if you are a Linux kernel expert and _fully_ understand all of the dependencies, or if the change is minor and you are _highly confident_ there are no dependencies.
+
+If you wish to use `menuconfig`, which figures out dependencies for you, you will need an environment in which to run it. Fortunately, the linuxkit project's kernel compile process already sets one up for you.
+To get an appropriate environment:
+
+1. `cd kernel/`
+2. Run a build for your desired kernel series, e.g. `make build_4.9.x ORG=foo HASH=bar`
+3. When you see the output from `make defconfig && make oldconfig` complete, hit `Ctrl-C` to stop the build
+4. Note the hash from the intermediate container. That intermediate container has all of the tools and source in it, and can be used to build.
+5. Get a shell in that intermediate container, mounting the current directory in: `docker run -it --rm -v ${PWD}:/src <hash> sh`
+
+This will give you a read-to-run kernel build environment, with all of the config files in `/src/`.
+
+For the output of step 4, e.g.:
+
+```
+Step X/29 : COMMAND
+ ---> b2a4a976d661
+```
+
```

Comment:
there are a few steps missing here. Mainly how to run `menuconfig` and then how to copy `.config` back to the source. They are kinda obvious, but given the rest of this addition is very detailed it's odd that the final steps are missing.

Also, this process will change quite a bit very soon (once https://github.com/linuxkit/linuxkit/tree/master/projects/kernel-config has been updated and merged)